src/freenet/client/async/SplitFileFetcherStorage.java

```diff
@@ -1144,6 +1149,9 @@ public boolean run(ClientContext context) {
                     if(succeeded) return false;
                     succeeded = true;
                 }
+                synchronized(cooldownLock) {
+                    overallCooldownWakeupTime = -1;
```

Comment:
A restructuring to avoid using a magic value would probably be extensive, so I won't propose doing that right now. What do you think of a FETCH_FINISHED constant that's -1?